docs/src/components/TypeScriptPlayground/impl.tsx

```diff
@@ -0,0 +1,334 @@
+        void tsRef.current.getEmitOutput(`file:///${rootDirectory}/${activeFileName}`).then((output) => {
+            const newData = Object.fromEntries(output.outputFiles.map((file) => [file.name, file.text]));
+            setEmittedFiles({ ...emittedFiles, ...newData });
+        });
+    }, [tsRef, emittedFiles]);
+
+    useEffect(() => {
+        if (isEditorReady) {
+            const activeUri = monacoRef.current.Uri.parse(`file:///${rootDirectory}/${activeFileName}`);
+            const model = monacoRef.current.editor.getModel(activeUri);
+            editorRef.current.setModel(model);
+        }
+    }, [isEditorReady, activeFileName]);
+
+    // Styling
+
+    const { colorMode } = useColorMode();
+    useUpdate(
+        () => {
+            monacoRef.current?.editor.setTheme(asMonacoTheme(colorMode));
+        },
+        [colorMode],
+        isEditorReady,
+    );
+
+    const [collapsed, setCollapsed] = useState<boolean>(true);
+    const toggleHeight = useCallback(() => {
+        setCollapsed(!collapsed);
+    }, [collapsed]);
+
+    const height = collapsed ? defaultHeight : `calc(${defaultHeight} * 2)`;
+
+    return (
+        <div>
+            <div className={styles.header}>
+                {files
+                    .filter((file) => !file.hidden)
+                    .map((file) => {
+                        return (
+                            <div
+                                key={file.name}
+                                className={clsx(styles.fileTab, { [styles.active]: file.name === activeFileName })}
+                                onClick={() => {
+                                    setActiveFileName(file.name);
+                                }}
+                            >
+                                {file.name}
+                            </div>
+                        );
+                    })}
+                <div className={styles.headerRight}>
+                    <div className={styles.sizeToggleButton} onClick={toggleHeight}>
+                        <IconArrow style={{ transform: collapsed ? "rotate(90deg)" : "rotate(-90deg)" }} />
+                    </div>
+                </div>
+            </div>
+            <Editor
+                {...editorProps}
+                height={height}
+                wrapperProps={{
+                    className: styles.editor,
+                }}
+                options={{
+                    tabSize: 2,
+                    lineNumbers: "off",
+                    scrollBeyondLastLine: false,
+                    folding: false,
+                    bracketPairColorization: { enabled: false },
+                    minimap: {
+                        enabled: true,
+                        renderCharacters: false,
+                    },
+                }}
+                // path={activeFile.name}
+                // defaultLanguage={activeFile.language ?? "typescript"}
+                // defaultValue={activeFile.text}
```

Comment:
Lots of commented out code in this file, is that intentional?

Reply:
Ah thanks, definitely not intentional just forgot to clean it up